stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/repository/QueryChannelsImplRepositoryTest.kt

```diff
@@ -1,36 +1,107 @@
+                cids = listOf("cid1", "cid2"),
+                sort = QuerySort.Companion.desc(Channel::cid)
+            )
+        )
+
+        verify(dao).insert(
+            argThat {
+                this.query.cids == listOf("cid1", "cid2") &&
+                    sortInnerEntities.any { it.name == "cid" && it.direction == QuerySort.SortDirection.DESC.value }
+            }
+        )
+    }
+
+    @Test
+    fun `Given query channels spec in DB When select by id Should return not null result`() = runBlockingTest {
+        whenever(dao.select("id1")) doReturn randomQueryChannelsWithSorts(
+            id = "id1",
+            filterObject = Filters.contains("cid", "cid1"),
+            cids = listOf("cid1")
+        )
+
+        val result = sut.selectById("id1")
+
+        Truth.assertThat(result).isNotNull()
+        Truth.assertThat(result!!.filter).isInstanceOf(ContainsFilterObject::class.java)
+        Truth.assertThat((result.filter as ContainsFilterObject).fieldName).isEqualTo("cid")
+        Truth.assertThat(result.filter.value).isEqualTo("cid1")
+        Truth.assertThat(result.cids).isEqualTo(listOf("cid1"))
+    }
 
     @Test
-    fun testInsertAndRead() = runBlocking {
-        val queryChannelsSpec = QueryChannelsSpec(data.filter1, QuerySort())
-        queryChannelsSpec.cids = listOf("a", "b", "c")
-        repo.insertQueryChannels(queryChannelsSpec)
-        val fromDB = repo.selectQueryChannels(queryChannelsSpec)
-        Truth.assertThat(queryChannelsSpec).isEqualTo(fromDB)
+    fun `Given no row in DB with such id When select by id Should return null`() = runBlockingTest {
+        whenever(dao.select("id1")) doReturn null
+
+        val result = sut.selectById("id1")
+
+        Truth.assertThat(result).isNull()
     }
 
     @Test
-    fun testUpdate() = runBlocking {
-        val queryChannelsSpec = QueryChannelsSpec(data.filter1, QuerySort())
-        queryChannelsSpec.cids = listOf("a", "b", "c")
-        repo.insertQueryChannels(queryChannelsSpec)
-        queryChannelsSpec.cids = listOf("a", "b", "c", "d")
-        repo.insertQueryChannels(queryChannelsSpec)
-
-        val fromDB = repo.selectQueryChannels(queryChannelsSpec)
-        Truth.assertThat(fromDB).isEqualTo(queryChannelsSpec)
+    fun `Given query channels specs in DB When select by ids Should return according results`() = runBlockingTest {
+        whenever(dao.select(listOf("id1", "id2"))) doReturn listOf(
+            randomQueryChannelsWithSorts(
+                cids = listOf(
+                    "cid1",
+                    "cid2"
+                )
+            ),
+            randomQueryChannelsWithSorts(cids = listOf("cid2", "cid3"))
+        )
+
+        val result = sut.selectQueriesChannelsByIds(listOf("id1", "id2"))
+
+        Truth.assertThat(result.size).isEqualTo(2)
+        Assert.assertTrue(result.any { it.cids == listOf("cid1", "cid2") })
+        Assert.assertTrue(result.any { it.cids == listOf("cid2", "cid3") })
```

Comment:
Why didn't you use `Truth` in these two cases?

Reply:
Because it's not convenient. Result here is a list, Truth can offer only `contains` method where you should pass exactly equal object. Kotlin's `any` lets you pass some predicate

Reply:
I think that `Truth.assertThat(result.any { it.cids == listOf("cid2", "cid3") }).isTrue()` doesn't look so bad, but it depends on preferences :)